### PR TITLE
Final Review

### PR DESCRIPTION
**Pull request by PowerLevel9000**
# Booking-app

# [Backend Pr](https://github.com/enis-memic/booking_app_api/pull/35)

- [Project Requirements](https://github.com/microverseinc/curriculum-final-capstone/blob/main/projects/business_requirements.md)
- [Kanban board](https://github.com/enis-memic/booking_app_api/projects/2)

### Backend repository
- [Backend repository](https://github.com/ernestmusong/Booking-app-frontend)
- [Api Docs](https://booking-app-api-lmvm.onrender.com/api-docs/index.html)
### Authors
- @enis-memic 
- @ernestmusong 
- @PowerLevel9000 
- @Habtamu210 

## Key Features
- login/signup
- get cars for reservation
- reservations and cacelations
- add more cars and delete them only for authorize admin 

## Important Steps
- create `.env` file at the root dir and add `port=8000` line in it

##### To see app running locally

- Please switch base url in the frontend app from rendered url to our localhost URL in order to see our app running locally with backend with their prequesties `http://localhost:3000/` for backend using `rails s` and http://localhost:8000/ for frontend

## Happy Devloping 

CelebrationGIF

### CODE REVIEW

File: src/components/Button.jsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import '../styles/Button.css';
+
+function Button({ text, className, fun }) {
```
Comment (Meltrust) [inline src/components/Button.jsx]: - [ ]  Please, kindly refactor this and all other functions in your project to ES6 arrow functions, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const Button = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

You can do a search in Vscode for the "function" word, and correct as needed.  Shouldn't take you more than 5-10 min so don't worry:

image


That way, your code will meet the latest standards. 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @PowerLevel9000,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Nice code organization ✔️ 
-  App is working well✔️ 
-  Good readme ✔️ 
-  Nice frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/components/Button.jsx (hunk 1 above)
Comment (PowerLevel9000) [reply src/components/Button.jsx]: hi @Meltrust  
Thanks for the suggestions we are refactoring it

**Review by Meltrust — APPROVED** [review]
#### Hi @PowerLevel9000,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Functions are now ES6✔️ 
- Nice work✔️ 




Cheers and Happy coding!👏👏👏


------

**Comment by PowerLevel9000** [comment]
# Thank you 💪💪💪🙏🙏🙏

ThankYouSoMuchPrabhasGIF


hi @Meltrust ,

we wanted to express my gratitude for your willingness to take the time to review our code on GitHub. Your feedback and suggestions have been instrumental in improving the quality of our work, and we appreciate your insights and expertise in this area.

It's inspiring to have someone like you, who is knowledgeable and experienced, provide valuable feedback that helps us enhance our coding skills. we truly appreciate the time and effort you put into reviewing our code, and we are grateful for your constructive criticism.

If there are any areas that we need to work on, please don't hesitate to let us know. Iwe are open to hearing any suggestions or recommendations you may have. Your feedback is essential to our growth as a developer, and we value it tremendously.

Thank you again for your time and for sharing your expertise with me. we are looking forward to applying the insights gained from your review to my future projects.

Sincerely,
@PowerLevel9000

**Pull request merged**